### PR TITLE
Update istio to work with new nginx spoof image

### PR DESCRIPTION
**Pull request by brendarearden**
**Issue**
https://github.com/rancher/rancher/issues/30842

**Depedent PR**
https://github.com/rancher/istio-installer/pull/32

**Problem**
istioctl upgrade reaches out to external github to download manifests, which breaks in an airgapped environment. The istio-installer image has been updated to intercept that call and download the manifests from the istio-installer image. 

**Solution**
The chart has been upgraded to pass in the proper environment variables for the new image mirror nginx server. It has also been updated to to allow a user to mount a manifest tar to the job pod to be served up by the nginx container to cover instances where an manifest tar wasn't bundled with the istio-installer

### CODE REVIEW

File: packages/rancher-istio/charts/values.yaml
Comment (jiaqiluo) [inline packages/rancher-istio/charts/values.yaml:5]: Should we use a rancher's repo here?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: packages/rancher-istio/charts/values.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/values.yaml:5]: Yes, I will switch this over once the new image is tagged .

File: charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-uninstall-job.yaml
Comment (jiaqiluo) [inline charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-uninstall-job.yaml:49]: 1. We are not checking the value of `.Values.global.rbac.pspEnabled` at this place in the `istio-install-job.yaml` file. Should we add it there? 
2. In both files, the `securityContext` at the container lever will be applied anyway, which makes the check useless. I think we should either remove the `securityContext` at the container level or move the checks there.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (cbron) [inline packages/rancher-istio/charts/templates/istio-install-job.yaml:60]: hostPath is not a good idea for a few reasons, one is that the tar won't exist on every node and we don't yet have nodeSelectors. Lets drop the volume part for now and release not that this only supports upgrading from versions available from rancher 2.5.0-2.5.9

File: packages/rancher-istio/charts/templates/istio-uninstall-job.yaml
Comment (cbron) [inline packages/rancher-istio/charts/templates/istio-uninstall-job.yaml:49]: Why do we wrap this one in conditional but not the above one ?

File: charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-uninstall-job.yaml
Comment (brendarearden) [reply charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-uninstall-job.yaml]: I have moved the security context to the container level for this and am no longer doing the rbac.pspEnabled check as it should always run as user 101

File: charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-install-job.yaml
Comment (brendarearden) [reply charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-install-job.yaml]: Yeah I thought that was weird too - I have removed it and am now just relying on the values.yaml to be set

File: packages/rancher-istio/charts/values.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/values.yaml]: this is now set to false

File: packages/rancher-istio/charts/values.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/values.yaml]: Updated the wording with your suggestion.

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml:18]: That is correct, this now replaces the need for the /etc/host. I have now wrapped it in an if statement for mirror enabled

File: packages/rancher-istio/charts/templates/base-config-map.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/base-config-map.yaml]: I have reverted this change. I had originally changed it so that I could store more files in the `configs/`

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml]: https://github.com/rancher/rancher/issues/34858 I have opened an issue here to cover this scenario and I am removing the hostPath change from the chart.

File: packages/rancher-istio/charts/templates/istio-uninstall-job.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-uninstall-job.yaml]: Originally the security context was only set when PSP was enabled, but now it needs to be on all the time - I have removed the job level security context and I am only keeping the container level securityContext so this is not longer an issue.

**Review by jiaqiluo — DISMISSED** [review]

File: packages/rancher-istio/charts/templates/istio-uninstall-job.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/templates/istio-uninstall-job.yaml:34]: same comment as above re: choice of user / group

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/templates/istio-install-job.yaml:45]: Should this be configurable (and also update the PSP)? Why was it set to 101?

Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml:45]: nit: why not just leave this where it originally was under serviceAccountName?

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/templates/istio-install-job.yaml:29]: Why did we remove the default?

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/templates/istio-install-job.yaml:31]: should a default be introduced here too?

File: packages/rancher-istio/charts/values.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/values.yaml:64]: 👍  this changes istio to have the same default as monitoring

**Review by aiyengar2 — COMMENTED** [review]
needs to change image before approval

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml:31]: default cannot be introduced here because the value is used for an if check else where and I don't want inconsistencies in how that evaluates. Its also set by default in the values.yaml so unless the user removes the value all together then it will be set.

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (cbron) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml:29]: Because its set in the values.yaml, whats point in keeping it both places ?

Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml]: this is added back in with the most recent commits - I originally removed it because we are setting this by default in the values.yaml so doesn't make sense to also have a default value - but it also doesn't hurt in this situation.

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml]: Moved back to global job setting instead of per contian. This should not be configurable because the istio-installer image does not run as root so I do not want people to attempt to run the container as root. The user in the istio-installer is a 101 user.

File: packages/rancher-istio/charts/templates/istio-uninstall-job.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-uninstall-job.yaml]: see answer to other security context question.

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml]: https://github.com/nginxinc/docker-nginx/blob/d496baf859613adfe391ca8e7615cc7ec7966621/mainline/alpine/Dockerfile#L17 additional context of how the user is generated - this is from the alpine image, but its the same across all nginx user setups.

**Review by cbron — COMMENTED** [review]
lgtm, but wait for @aiyengar2 and need to tag image

**Comment by brendarearden** [comment]
Image has been updated to the most recent istio-installer tag and is ready for final review.

File: charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-uninstall-job.yaml
Comment (brendarearden) [reply charts/rancher-istio/rancher-istio/100.0.1+up1.10.4/templates/istio-uninstall-job.yaml]: Moved back to job context instead of container context.

**Review by jiaqiluo — APPROVED** [review]

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml]: ah yes this makes sense. Thanks for the clarification!

File: packages/rancher-istio/charts/templates/istio-install-job.yaml
Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-install-job.yaml:31]: sounds good to me. I don't think dropping it is that much of a concern.

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Pull request merged**
